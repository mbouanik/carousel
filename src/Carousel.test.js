import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

it("works when you click on the right arrow", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />,
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]'),
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]'),
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]'),
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]'),
  ).toBeInTheDocument();
});

it("should test snapshot and smoke test", () => {
  const { asFragment, getByText, getByTestId } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />,
  );
  expect(asFragment).toMatchSnapshot();
});

it("should display next img", () => {
  const { getByText, getByTestId } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />,
  );
  const h4 = getByText("testing image 1");
  const right_btn = getByTestId("right-btn");
  const left_btn = getByTestId("left-btn");

  fireEvent.click(right_btn);
  expect(h4).toHaveTextContent("testing image 2");
  expect(h4).not.toHaveTextContent("testing image 1");

  fireEvent.click(left_btn);
  expect(h4).not.toHaveTextContent("testing image 2");
  expect(h4).toHaveTextContent("testing image 1");
});

it("should not overflow the array of images or negative index", () => {
  const { getByText, getByTestId } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />,
  );
  const h4 = getByText("testing image 1");
  const right_btn = getByTestId("right-btn");
  const left_btn = getByTestId("left-btn");

  fireEvent.click(left_btn);
  expect(h4).toHaveTextContent("testing image 1");
  fireEvent.click(right_btn);
  expect(h4).toHaveTextContent("testing image 2");
  expect(h4).not.toHaveTextContent("testing image 1");
  fireEvent.click(right_btn);
  expect(h4).toHaveTextContent("testing image 3");
  expect(h4).not.toHaveTextContent("testing image 2");

  fireEvent.click(right_btn);
  expect(h4).toHaveTextContent("testing image 3");
});
