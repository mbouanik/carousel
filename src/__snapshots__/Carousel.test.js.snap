// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test snapshot and smoke test 1`] = `[Function]`;
